mis/data/graphs.py

```diff
@@ -538,3 +540,39 @@ def ingest(self, graph: NXWithPos, device: pl.devices.Device, id: int) -> BaseGr
         pyg.pos = torch.tensor(positions, dtype=torch.float)
 
         return BaseGraph(id=id, device=device, data=pyg)
+
+
+@dataclass
+class DIMACSDataset:
+    instance: MISInstance
+    solutions: list[list[int]]
+
+
+def load_dimacs(path: str) -> DIMACSDataset:
+    """
+    Load a DIMACS file and return a DIMACSDataset.
+
+    Args:
+        path (str): Path to the DIMACS file.
+
+    Returns:
+        DIMACSDataset: An instance of DIMACSDataset.
+    """
+    with open(path, 'r') as f:
+        lines = f.readlines()
+
+    # Parse the graph from the DIMACS format
+    edges = []
+    for line in lines:
+        if line.startswith('p'):
+            continue  # Skip problem line
+        if line.startswith('e'):
+            parts = line.split()
+            edges.append((int(parts[1]), int(parts[2])))
+
+    graph = nx.Graph()
+    graph.add_edges_from(edges)
+    instance = MISInstance(graph)
+    solver = MISSolver(instance, SolverConfig())
+    solutions = [e.nodes for e in solver.solve().result()]
```

Comment:
Ah, I realize that the issue description wasn't clear. We don't want to call the solver here. It's just that some DIMACS files contain a solution (or perhaps more than one solution, I haven't checked). `solutions` should be the solution lines.

@RolandMacDoland do you have any such file at hand, for testing?

Reply:
Do you mean solutions ? If so no. Sometimes DIMACS files contain them that I use in tests. However, there is no guarantee that the current solver returns the same set of node labels.

Reply:
@RolandMacDoland I meant, do you have an example file that contains the solution? To confirm the syntax.

Reply:
@y-richie-y Could you fix this?

Reply:
Sure. What do you want me to do when there is no solution line?